### ISSUE TITLE
unexpected result can happen if a zero ORAT history controlled well is located in the water layer

### ISSUE THREAD

**Issue by GitPaean** [OPM/opm-simulators#4295]
Since there is no oil in the water layer anyway, if we specify the history well is under ORAT control, then there is basically no RATE constraint for this well and the only constraint will be the BHP constraint, it can be 1atm.  Then you might see the well is producing huge amount of water under 1 atm BHP control. 

Here, it is describing the specific setup that we had problem with, and it is a general problem. 

Basically, we should make sure the well is STOPped for this kind of setup.

**Comment by OPMUSER** [comment]
@GitPaean I think this is the correct behavior, so may be there is something else amiss.

Is the WHISTCTL keyword in the input deck? And if it is, has `WHISTCTL(END)` been set to YES?. According to the manual we don't support ` WHISTCTL(END) = YES`, only the NO option.

Also, is the BHP defined on the WCONHIST and/or WCONPROD keyword?

**Comment by OPMUSER** [comment]
@GitPaean Also is there a `WELTARG` keyword in the input deck?

**Comment by GitPaean** [comment]
Hei, there is no WELTARG. 

There is `WHISCTL`, 
```
 982 WHISTCTL 
 983  RESV /
 984 -- ORAT /
 985   
```

Then maybe the real bug is why the `WHISCTL` is not working here, it is defined after the `WCONHIST` for this well? Not sure what you mean by `WHISTCTL(END)`. 

BHP is not defined with the WCONHIST keyword, so it is the default one, 1atm.

**Comment by OPMUSER** [comment]
> Then maybe the real bug is why the WHISCTL is not working here, it is defined after the WCONHIST for this well

Yes, you are right, **all** wells should be on RESV control - so check the control mode of all the other wells as well.

> Not sure what you mean by WHISTCTL(END).

The second parameter on the  `WHISTCTL` keyword, `WHISTCTL(END)`, determines if the run should terminate if it switched to BHP control (YES), or not (NO). We only support the NO option.

**Comment by GitPaean** [comment]
> Yes, you are right, all wells should be on RESV control - so check the control mode of all the other wells as well.

I will check that when coming to fix it.  At the moment, there are too many things going on here.

**Comment by GitPaean** [comment]
> Yes, you are right, all wells should be on RESV control - so check the control mode of all the other wells as well.

Hei, @OPMUSER, does `WHISTCTL` change the control for literally all the wells for the whole simulation, or only the `WCONHIST` entered after the `WHISCTL` keyword? When I read, I got the impression is that only the keywords `WCONHIST` entered after the `WHISCTL` will be changed?

**Comment by atgeirr** [comment]
Can this be closed now, or are we still vulnerable to this?
